### ISSUE TITLE
RadioGroupField inside container with display:grid causes UI to break

### ISSUE THREAD

**Issue by Alex-Github-Account** [aws-amplify/amplify-ui#3165]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Other

### How is your app built?

Create React App

### What browsers are you seeing the problem on?

Chrome

### Please describe your bug.

when RadioGroupField placed inside container which has css
display:grid 
and then any of the Radio inside said RadioGroupField clicked
then the effect happens which looks like all the elements inside display:grid container receved position:absolute
I researched, and the cause seems to be 
position: absolute;
of CSS class 
.amplify-visually-hidden 
when this CSS rule disabled with
[class*=amplify].amplify-visually-hidden{position:unset;}
everything becomes normal (and also, small 1x1px  'input'  can be seen to appear unless additionally
opacity: 0
applied)

To clearify, no OnCange is set, as well as no any events at all. This seems to be somehow connected with element state change. I thing so because if radio clicked and bug happened, and then hidden 'input' element is deleted via browser tools, and then deletion revert back by pressing ctrl+z, evetyhing comes back to normal - I suspect, as a result of element statr (like :focused, :active, etc) been reset upon element been re-added to dom.

Forgot to add, this all should be inside 'dialog' tag. Please wait jsfiddle or video.

### What's the expected behaviour?

Any amplify styles should not affect UI
RadioGroupField should work, be clicked without consequences

### Help us reproduce the bug!

I will do jsfiddle later. I am writing this bug to ouline the issue and maybe others who expiriencing same issue will support.
 

### Code Snippet

```typescript
 <fieldset style="display:grid"> 
             <RadioGroupField
                label=""
          name="buggy"
                >
               <Radio value="0" >Value 1</Radio>
               <Radio value="1" >Value 2</Radio>
               <Radio value="2" >Value 3</Radio>
            
               </RadioGroupField>
</fieldset>
```

### Additional information and screenshots

will do video

**Comment by reesscot** [comment]
Hi @Alex-Github-Account 
Sorry to hear you're having this problem, but I'm unable to reproduce this issue following the instructions listed above. See:
https://stackblitz.com/edit/react-ts-xtzcwf?file=App.tsx
Can you please provide a StackBlitz minimal reproduction of this issue?

**Comment by Alex-Github-Account** [comment]
Yes I will do. Issue is persistent on my side thus I'd be able to cut off irrelevant parts of page on which it is happening and put into fiddle. 
So far I need to finish the task I am working on, and turned out the CSS fix I provided does not fix bug entirely: the 'circle' of radio becomes non-circle (width smaller than height) on certain circumstance, thus I abandoned amplify radio and switched to one from another lib.  Let me some time and I'll populate your fiddle until it will start showing the issue.

**Comment by Alex-Github-Account** [comment]
> 

https://stackblitz.com/edit/react-ts-b2p5sy

scroll down and click any of the radios
I just pasted all the css from project inthere in one file so there is a mess, but please pay attention only to 
display: grid 
only which causes a bug
also apply my css fix from first post to see bug dissapear
also notice there is no 
:focus
:focus-within
:active
css selectors in the CSS, so whatever happens happens not because of some state-selector 
the distorted-circle proportions of 'radio' circle bug is not happening in that fiddle, I actually have no time to reproduce it in fiddle, I realised amplify ui have a long way to go and switched to better ui lib,

**Comment by Alex-Github-Account** [comment]
does this helps? reproduceable?

**Comment by Alex-Github-Account** [comment]
and thats why no one likes doing that jsfiddles when opening a bug - after a time spent on doing all of the work of finding and isolating a bug, then... nothing happens. like, no one seems to care.

**Comment by Alex-Github-Account** [comment]
realized, for bugreport to be on my portfolio, I reopen it back.

**Comment by reesscot** [comment]
@Alex-Github-Account When I looked at this before, it was difficult to diagnose with such a large reproduction example (there's a lot of CSS in the stackblitz example that could have been causing the issue). However, if you delete all the extra `<br/>` tags in your example, you'll see that the issue you're seeing goes away. This does not appear to be a bug with the radio button. 

![CleanShot 2023-01-18 at 14 10 00](https://user-images.githubusercontent.com/6165315/213295515-89b25308-c6aa-4a33-994d-b7a4a88f3cca.gif)
Here's the forked version with only the `<br>` tags removed:
https://stackblitz.com/edit/react-ts-ntnzzt?file=style.css,App.tsx

**Comment by Alex-Github-Account** [comment]
> @Alex-Github-Account When I looked at this before, it was difficult to diagnose with such a large reproduction example (there's a lot of CSS in the stackblitz example that could have been causing the issue). However, if you delete all the extra `<br/>` tags in your example, you'll see that the issue you're seeing goes away. This does not appear to be a bug with the radio button.
> 
> ![CleanShot 2023-01-18 at 14 10 00](https://user-images.githubusercontent.com/6165315/213295515-89b25308-c6aa-4a33-994d-b7a4a88f3cca.gif) [ ![CleanShot 2023-01-18 at 14 10 00](https://user-images.githubusercontent.com/6165315/213295515-89b25308-c6aa-4a33-994d-b7a4a88f3cca.gif) ](https://user-images.githubusercontent.com/6165315/213295515-89b25308-c6aa-4a33-994d-b7a4a88f3cca.gif) [ ](https://user-images.githubusercontent.com/6165315/213295515-89b25308-c6aa-4a33-994d-b7a4a88f3cca.gif) Here's the forked version with only the `<br>` tags removed: https://stackblitz.com/edit/react-ts-ntnzzt?file=style.css,App.tsx

 Once again, please read what I write
>difficult to diagnose

I did that for you read https://github.com/aws-amplify/amplify-ui/issues/3165#issuecomment-1352161151 qoute "'display: grid' only which causes a bug"

> there's a lot of CSS i

read https://github.com/aws-amplify/amplify-ui/issues/3165#issuecomment-1352161151 qoute "there is no
:focus
:focus-within
:active
css selectors in the CSS", means **none** of my css address state change. no state-dependent css --> **none of my css applied specifically when user inteacts (clicks)**

> if you delete all the extra <br/> tags

seriously? didnt it cross your mind that 'all the extra br' is here to serve as placeholder for content that otherwice would be there? Do you prefer me to bloat up example with some real-world form controls instead of br? here we go (bug still appears, Chrome 109):

https://stackblitz.com/edit/react-ts-eeqwty?file=App.tsx

it is not hard for me to paste in few kilobytes of irrelevant form controls if you dont like br, you are welcome :D

**Comment by Alex-Github-Account** [comment]
another month gone, radio control still breaks up entrie interface if used. 

Impressive both: 
- ability to create such a component that somehow breaks whole chain of it's parents. 
- response time to fix that 
- 
 Thumbs up!

**Comment by calebpollman** [comment]
Hi @Alex-Github-Account our apologies on the frustration you have experienced with this issue.

I've been trying to work through the issue you are having using the provided stackblitz examples, and am unable to reproduce the buggy behavior unless the `<br />` elements are present in the first example, and am having a much harder time attempting to debug the second example due it's size and complexity.

This does not mean that there is not a bug within the `RadioGroupField`, but unless we have a small, scoped example without the `<br />` elements and minimal CSS, it is hard to determine where the issue lies if it does indeed exist.

**Comment by dbanksdesign** [comment]
@Alex-Github-Account I spent a few hours poking around at it this morning and I have a few solutions, but I don't quite know what the root cause is. For some reason the `position: absolute` style on the `amplify-visually-hidden` class is causing the issue. Changing that to `position: fixed` fixes the problem. Also, a quick change to your code that would work as well would be to add `position: relative` somewhere up the DOM tree, like on your `main` element. I created a minimally (as much as I could) reproducible example:

https://stackblitz.com/edit/react-9ghimk?file=src%2FApp.js

**Comment by Alex-Github-Account** [comment]
Thanks. 
 it's only part of a problem. The radio element becomes oval (not circle) sometimes but it didn't included in this example for simplicity. 
So my take on the root cause: instead of bootstrap-way (where native radio input got styled via css), someone decided that it would be cool kid move to hide native input element completely (hello accessibility) and 'paint' fake ui element with div's/svg/whatever from scratch. As if native css-styled element not works good enough. 
I ended up using bootstrap and did not expirienced not a single problem with it for obvious reason: they just don't break what works. and native radio element kind of... works. (and can be styled well).  
The rest (like, adding and removing the but flair for this thread, ui components magically affecting parents, month+ to reply, br's joke, etc) is just a funny thing I'd be refer my friends too. Some parts of amplify ui just need to be dumped and engineered the proper way.

**Comment by Alex-Github-Account** [comment]
@reesscot the bug flair hurts your feelings after BR's incident? or an _UI component that breaks entrie page_ is not a bug?

**Comment by dbanksdesign** [comment]
@Alex-Github-Account there was a previous bug where the radio button would get squished horizontally. That bug should have been fixed in #3074. I don't see that bug still appearing in the linked examples. If you are still seeing that issue please let me know. Your point about styling the native input versus hiding it and re-creating it is fair. Having a custom element for the radio button does allow for more styling than the native input and is a fairly common technique used in other UI libraries. 

The fix for us would be to add `position: relative` to `amplify-radio` class.

**Comment by dbanksdesign** [comment]
We pushed a fix that will go out in our release tomorrow.
